athenahealth/appointments.go

```diff
@@ -622,6 +622,52 @@ func (h *HTTPClient) BookAppointment(ctx context.Context, patientID, apptID stri
 	return out[0], nil
 }
 
+type UpdateBookedAppointmentOptions struct {
+	// New appointment type ID for this appointment.
+	AppointmentTypeID *string `json:"appointmenttypeid"`
+	// New department ID for this appointment.
+	DepartmentID *string `json:"departmentid"`
+	// New provider ID for this appointment.
+	ProviderID *string `json:"providerid"`
+	// New supervisingprovider ID for this appointment.
+	SupervisingProviderID *string `json:"supervisingproviderid"`
+}
+
+// UpdateBookedAppointment
+// PUT /v1/{practiceid}/appointments/booked/{appointmentid}
+// https://docs.athenahealth.com/api/api-ref/appointment-booked#Appointment-Booked
+func (h *HTTPClient) UpdateBookedAppointment(ctx context.Context, apptID string, opts *UpdateBookedAppointmentOptions) error {
+	form := url.Values{}
+
+	if opts.AppointmentTypeID != nil {
+		form.Add("appointmenttypeid", *opts.AppointmentTypeID)
+	}
+
+	if opts.DepartmentID != nil {
+		form.Add("departmentid", *opts.DepartmentID)
+	}
+
+	if opts.ProviderID != nil {
+		form.Add("providerid", *opts.ProviderID)
+	}
+
+	if opts.SupervisingProviderID != nil {
+		form.Add("supervisingproviderid", *opts.SupervisingProviderID)
+	}
+
+	outNum := 0
+	res, err := h.PutForm(ctx, fmt.Sprintf("/appointments/booked/%s", apptID), form, &outNum)
+	if err != nil {
+		if strings.Contains(err.Error(), "cannot unmarshal string into Go value of type int") {
+			return BodyToErrorString(res.Body)
+		}
+
+		return err
+	}
+
+	return nil
+}
+
```

Comment:
We can simplify this (and avoid error string checking) to use `NumberString` and directly compare against the successful value here

```suggestion
	var statusRes NumberString
	_, err := h.PutForm(ctx, fmt.Sprintf("/appointments/booked/%s", apptID), form, &statusRes)

	if err != nil {
		return err
	}

	if string(statusRes) != successfulBookingResponse {
		return errors.New(string(statusRes))
	}

	return nil
```

Reply:
this is awesome! thanks ellie that definitely saves a lot of code.